src/System.Private.CoreLib/src/System/Runtime/CompilerServices/ConditionalWeakTable.cs

```diff
@@ -424,6 +433,22 @@ internal bool Remove(TKey key)
             }
 
             //----------------------------------------------------------------------------------------
+            // Removes all entries from the table.
+            //
+            // Precondition:
+            //      Must hold _lock.
+            //----------------------------------------------------------------------------------------
+            internal void Clear()
+            {
+                // Remove all handles (as in Remove, just by setting the hashcode and leaving
+                // actual removal up to the finalizer).
+                for (int i = 0; i < _firstFreeEntry; i++)
```

Comment:
Looks like our comments crossed but we landed in the same place.

I submitted a PR to CoreCLR to just replace the Container, I updated this PR to do the same, and I opened an issue in CoreCLR to look at improving Remove as well.